src/main/java/com/soen343/gms/controller/UserController.java

```diff
@@ -98,11 +122,25 @@ public ModelAndView archivedjobs(){
         return model;
     }
   
-    @RequestMapping(value={"/home/job/job1"}, method=RequestMethod.GET)
-    public ModelAndView jobpage(){
+    @RequestMapping(value={"/home/job/{id}"}, method=RequestMethod.GET)
+    public ModelAndView jobpage(@PathVariable("id") long id){
         ModelAndView model = new ModelAndView();
-        model.setViewName("job/jobpage");
+        Job job = jobService.getJobById(id);
+        model.setViewName("job/" + id);
```

Comment:
Wait, will this actually go looking for a specific page in the resources directory? Not sure setting the view name using the ID is correct. I think you should actually just refer to the "job/jobpage" and simply pass in the corresponding job object that was actually asked for